distributed_training/utils/state_loader.py

```diff
@@ -51,39 +39,44 @@ def set_loading_state(self, is_loading, epoch=None):
                 self._last_loaded_epoch = epoch
 
 
-def load_model_optimizer_gradient_averager(self, epoch):
+def load_model_optimizer_gradient_averager(self, model_name, epoch):
     bt.logging.info(
         f"CPU Memory Before Loading State {psutil.virtual_memory().available / 10**9} GB"
     )
     # Delete existing model
     if hasattr(self, "model"):
-        del self.model.model.transformer.wte.weight
-        del self.model.model.transformer.wte.norm
-        del self.model.model.transformer.wpe.weight
-        del self.model.model.transformer.wpe.norm
-        del self.model.model.transformer.wte
-        del self.model.model.transformer.wpe
+        if hasattr(self.model.model.transformer.wte, "weight"):
+            del self.model.model.transformer.wte.weight
+        if hasattr(self.model.model.transformer.wte, "norm"):
+            del self.model.model.transformer.wte.norm
+        if hasattr(self.model.model.transformer.wpe, "weight"):
+            del self.model.model.transformer.wpe.weight
+        if hasattr(self.model.model.transformer.wpe, "norm"):
+            del self.model.model.transformer.wpe.norm
+        if hasattr(self.model.model.transformer, "wte"):
+            del self.model.model.transformer.wte
+        if hasattr(self.model.model.transformer, "wpe"):
+            del self.model.model.transformer.wpe
         del self.model
         gc.collect()
         torch.cuda.empty_cache()
 
     # Load a new model
     self.model = (
         AutoModelForCausalLM.from_pretrained(
-            self.config.neuron.model_name, revision=str(epoch), trust_remote_code=True
+            model_name, revision=str(epoch), trust_remote_code=True
         )
         if epoch
-        else AutoModelForCausalLM.from_pretrained(
-            self.config.neuron.model_name, trust_remote_code=True
-        )
+        else AutoModelForCausalLM.from_pretrained(model_name, trust_remote_code=True)
     )
     # Move the model to the appropriate device
     self.model = self.model.to(self.device)
+    self.model.config.block_list = []
 
-    # Delete any historic model references in GlobalOptimManager
+    # Delete any historic model references in GlobalOptimManager # TODO Do we still need this in here?
```

Comment:
@jundi69 you're right we most likely don't need this now but I wouldn't mind keeping this in for now in case there is  ever an explosion in model weights stored in the cache which could lead to OOM errors.